_posts/2020-11-30-dataops-dynamodb-to-snowflake.md

```diff
@@ -0,0 +1,378 @@
+
+### Integrating with Data Warehouses
+Despite providing flexibility for transactional data users, NoSQL tables often present challenges for analytics teams. It can be difficult to transform semi-structured data to the relational table format found in data warehouses,  or make sense of sparse data patterns commonly used in single table designs.
+
+In this article I will to show you how to successfully integrate DynamoDB data with [Snowflake](https://www.snowflake.com), in a **cost-effective** and **cloud native** way. We will tackle both the sparse data and semi-structured data challenges, using plain old SQL.
+
+### What has changed?
+
+Until very recently, getting bulk data out of AWS DynamoDB typically required the use of either specialised vendor or cloud provider tools (e.g. AWS Data Pipeline), or invest a lot of your own engineering effort to extract the data yourself. In any case, there was no out-of-the-box feature that could dump a DynamoDB table for export, without having a performance, cost or engineering impact. 
+
+In November 2020, AWS DynamoDB has released a new feature to [export the contents of a table to S3](https://aws.amazon.com/blogs/aws/new-export-amazon-dynamodb-table-data-to-data-lake-amazon-s3/) without performance impact, leveraging *Point-In-Time-Recovery* (PITR). The cost for this serverless feature is based only on the volume of data that you export, priced at $0.114 per GB in the AWS Sydney region.
+
+This feature - which many would say is long overdue - may be a very welcome feature for those working with data lakes and data warehouses, as it allows for very simple cloud native integrations with your DynamoDB data:
+* **Snowflake** has excellent support for [External Tables](https://docs.snowflake.com/en/user-guide/tables-external-intro.html) that read directly from external stages as well as [table hydration from external stages](https://docs.snowflake.com/en/user-guide/data-load-s3-copy.html) for better query performance
+
+* **Amazon Redshift** has good support for [integrating data from DynamoDB](https://docs.aws.amazon.com/redshift/latest/dg/t_Loading-data-from-dynamodb.html), however, this latest feature may allow teams leveraging older methods to consolidate and simplify their data access patterns, whilst also eliminating load from their operational tables
+
+At the time of writing, the new export functionality is only available in the AWS DynamoDB V2 console, though it is expected this V2 console will shortly become the default, once it is feature complete. It is also available in SDK libraries and using the AWS Command Line Interface (CLI).
```

Comment:
What about CloudFormation, can you schedule exports via cloudwatch events etc? Or do you need to write a lambda function that uses the SDK?

Reply:
yeah it's a lambda in the repo i haven't pushed yet - will do a separate PR for that one